### PR TITLE
Add security policy

### PR DESCRIPTION
**Pull request by martincostello**
Add security policy documentation for GitHub.

Based on ASP.NET Core's.

Feedback on the wording aside, just needs the email address adding.

See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/issues/713#issuecomment-1232718044.

### CODE REVIEW

**Comment by martincostello** [comment]
Will copy this to the OpenID repo once it's merged.

File: .github/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+## Supported Versions
+
+Only versions of the providers that target versions of ASP.NET Core that are in support by Microsoft are supported by the maintainers.
+
+The .NET Core and ASP.NET Core support policy, including currently supported versions, can be found at the [.NET and .NET Core Support Policy page](https://dotnet.microsoft.com/platform/support/policy/dotnet-core).
+
+## Reporting a Vulnerability
+
+Security issues and bugs should be reported privately by emailing CHANGE_ME.
```
Comment (kevinchalet) [inline .github/SECURITY.md]: If you don’t mind adding my personal address, we can use contact@kevinchalet.com.

**Review by kevinchalet — APPROVED** [review]
Note: I would put SECURITY.md in the root folder so it's easier to find (it’s the default place if you use the “set up project security button").

File: .github/SECURITY.md (hunk 1 above)
Comment (martincostello) [reply .github/SECURITY.md]: That's fine with me - I just didn't want it to be my own personal email 😄

Comment (martincostello) [inline .github/SECURITY.md]:
```suggestion
Security issues and bugs should be reported privately by emailing [contact@kevinchalet.com](mailto:contact@kevinchalet.com).
```

**Comment by martincostello** [comment]
I'm happy to move the file if you want, but the `.github` folder is noted as a valid location in [the documentation](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository), so I figured I'd put it there as it's GitHub-specific and to de-clutter the root. I figured noting it in the README itself would be enough of a pointer to find it, and as a supported feature it  _should_ light up on the new issue page like it does for the aspnetcore repo (I believe).

**Comment by kevinchalet** [comment]
Makes sense, let's keep it as-is then :+1

Thanks!

**Comment by martincostello** [comment]
Yep, works 👍

image

**Pull request merged**